v4-client-js/src/clients/modules/post.ts

```diff
@@ -168,28 +218,62 @@ export class Post {
      * @description Calculate the default broadcast mode.
      */
     private defaultBroadcastMode(msgs: EncodeObject[]): BroadcastMode {
-      if (
-        msgs.length === 1 &&
-        (msgs[0].typeUrl === '/dydxprotocol.clob.MsgPlaceOrder' ||
-          msgs[0].typeUrl === '/dydxprotocol.clob.MsgCancelOrder')
-      ) {
-        const orderFlags = msgs[0].typeUrl === '/dydxprotocol.clob.MsgPlaceOrder'
-          ? (msgs[0].value as MsgPlaceOrder).order?.orderId?.orderFlags
-          : (msgs[0].value as MsgCancelOrder).orderId?.orderFlags;
-
-        switch (orderFlags) {
-          case OrderFlags.SHORT_TERM:
-            return Method.BroadcastTxSync;
-
-          case OrderFlags.LONG_TERM:
-          case OrderFlags.CONDITIONAL:
-            return Method.BroadcastTxCommit;
-
-          default:
-            break;
-        }
+      return this.areShortTermOrderMsgs(msgs) ? Method.BroadcastTxSync : Method.BroadcastTxCommit;
+    }
+
+    /**
+     * @description Calculate the default broadcast mode for a msg.
+     */
+    private defaultMsgBroadcastMode(msg: EncodeObject): BroadcastMode {
+      if (this.isShortTermOrderMsg(msg)) {
+        return Method.BroadcastTxSync;
+      } else {
+        return Method.BroadcastTxCommit;
       }
-      return Method.BroadcastTxSync;
+    }
+
+    /**
+     * @description Whether the msg is a short term placeOrder or cancelOrder msg.
+     */
+    private isShortTermOrderMsg(msg: EncodeObject): boolean {
+      if (msg.typeUrl === '/dydxprotocol.clob.MsgPlaceOrder' ||
+          msg.typeUrl === '/dydxprotocol.clob.MsgCancelOrder') {
+        const orderFlags = msg.typeUrl === '/dydxprotocol.clob.MsgPlaceOrder'
+          ? (msg.value as MsgPlaceOrder).order?.orderId?.orderFlags
+          : (msg.value as MsgCancelOrder).orderId?.orderFlags;
+
+        return orderFlags === OrderFlags.SHORT_TERM;
+      } else {
+        return false;
+      }
+    }
+
+    /**
+     * @description Whether the msg is a short term placeOrder or cancelOrder msg.
+     */
+    private isOrderMsg(msg: EncodeObject): boolean {
+      return (msg.typeUrl === '/dydxprotocol.clob.MsgPlaceOrder' ||
+          msg.typeUrl === '/dydxprotocol.clob.MsgCancelOrder');
+    }
```

Comment:
All dYdX specific typeUrls are included in `src/clients/constants.ts`. Should use those constants instead of hard coded strings here.